### PR TITLE
Add processors to javadoc classpath

### LINKED ISSUE #19: Javadoc classpath does not contain processor library
For Immutables, at least, when I generate Javadoc for classes that are annotated with Immutables annotations, I get warnings like this:

    javadoc: warning - Class Value.Immutable not found.

### PR DESCRIPTION
**Pull request by cjp39**
This fixes #19

### CODE REVIEW

File: src/main/groovy/org/inferred/gradle/ProcessorsPlugin.groovy [hunk 1]
```diff
@@ -34,12 +36,16 @@ class ProcessorsPlugin implements Plugin<Project> {
...
-          def path = project.files(config.getFiles(
-              { d -> true } as Spec<Object>)).getAsPath()
+          String path = getProcessors(project).getAsPath()
           project.compileJava.options.compilerArgs += ["-processorpath", path]
         }
       })
+      project.javadoc.dependsOn project.task('javadocProcessors', {
+        doLast {
+          Set<File> path = getProcessors(project).files
+          project.javadoc.options.classpath += path
```
Comment (markelliot) [inline src/main/groovy/org/inferred/gradle/ProcessorsPlugin.groovy]: this seems fine to me, but I'm not sure why you create a new task to do this vs. just adding an `afterEvaluate` block with the direct call.

File: src/test/groovy/org/inferred/gradle/ProcessorsPluginFunctionalTest.groovy [hunk 2]
```diff
@@ -211,6 +211,38 @@ public class ProcessorsPluginFunctionalTest {
...
+          return new AutoValue_MyClass(value);
+        }
+      }
+    """
+
+    String txt = GradleRunner.create()
+        .withProjectDir(testProjectDir.getRoot())
+        .withArguments("--info", "javadoc")
+        .build()
+        .getStandardError()
```
Comment (markelliot) [inline src/test/groovy/org/inferred/gradle/ProcessorsPluginFunctionalTest.groovy]: heads up, in 2.10 this gets replaced with this: https://docs.gradle.org/current/javadoc/org/gradle/testkit/runner/GradleRunner.html#forwardStdError(java.io.Writer)

**Comment by markelliot** [comment]
lgtm, can take as is, or hit the comments

File: src/main/groovy/org/inferred/gradle/ProcessorsPlugin.groovy (hunk 1 above)
Comment (cjp39) [inline src/main/groovy/org/inferred/gradle/ProcessorsPlugin.groovy]: afterEvaluate blocks are racey, and people tend to do configuration changes in them.

File: src/test/groovy/org/inferred/gradle/ProcessorsPluginFunctionalTest.groovy (hunk 2 above)
Comment (cjp39) [inline src/test/groovy/org/inferred/gradle/ProcessorsPluginFunctionalTest.groovy]: Thanks! I'll remember that for when we upgrade the test suite to run on 2.10 :)

**Pull request merged**